firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.remoteconfig.internal;
+
+import static com.google.firebase.remoteconfig.FirebaseRemoteConfig.TAG;
+
+import android.util.Log;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.remoteconfig.ConfigUpdateListener;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigException;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigRealtimeUpdateFetchException;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigRealtimeUpdateStreamException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class ConfigAutoFetch {
+
+  private static final int FETCH_RETRY = 3;
+
+  private final Set<ConfigUpdateListener> eventListeners;
```

Comment:
Updated. And yeah, with the next PR that includes the retry methods we would do this for them already. This exception would be more of an indication to the user that Realtime won't be able to work at all due to the stream issue. This may enable them to revert to base RC as a fallback or some other mitigation where they are not reliant on realtime.